packages/frontend-api/src/Model/Resolver/Products/Search/ProductSearchQuery.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\Resolver\Products\Search;
+
+use Overblog\GraphQLBundle\Definition\Argument;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrontendApiBundle\Component\Validation\PageSizeValidator;
+use Shopsys\FrontendApiBundle\Model\Product\Connection\ProductConnection;
+use Shopsys\FrontendApiBundle\Model\Product\Filter\ProductFilterFacade;
+use Shopsys\FrontendApiBundle\Model\Resolver\AbstractQuery;
+
+class ProductSearchQuery extends AbstractQuery
+{
+    /**
+     * @param \Shopsys\FrontendApiBundle\Model\Resolver\Products\Search\ProductSearchResultsProviderResolver $productSearchResultsProviderFacade
+     * @param \Shopsys\FrontendApiBundle\Model\Product\Filter\ProductFilterFacade $productFilterFacade
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     */
+    public function __construct(
+        protected readonly ProductSearchResultsProviderResolver $productSearchResultsProviderFacade,
```

Comment:
class type name vs property name mismatch, could you please rename the property to `$productSearchResultsProviderResolver`?